sigar_linux.go

```diff
@@ -317,24 +317,18 @@ func (self *NetIfaceList) Get() error {
 		mtuFile := fmt.Sprintf("%v/class/net/%v/mtu", Sysd, ifaceList[i].Name)
 		macFile := fmt.Sprintf("%v/class/net/%v/address", Sysd, ifaceList[i].Name)
 		linkStatFile := fmt.Sprintf("%v/class/net/%v/carrier", Sysd, ifaceList[i].Name)
-		mtu, err := ioutil.ReadFile(mtuFile)
-		if err == nil {
-			ifaceList[i].MTU, _ = strtoull(string(mtu))
-		}
-		macAddr, err := ioutil.ReadFile(macFile)
-		if err == nil {
-			ifaceList[i].Mac = string(macAddr)
-		}
-		linkStat, err := ioutil.ReadFile(linkStatFile)
-		if err == nil {
-			switch string(linkStat) {
-			case "0":
-				ifaceList[i].LinkStatus = "DOWN"
-			case "1":
-				ifaceList[i].LinkStatus = "UP"
-			default:
-				ifaceList[i].LinkStatus = "UNKNOWN"
-			}
+
+		ifaceList[i].MTU = readUint(readFileLine(mtuFile))
+		ifaceList[i].Mac = readFileLine(macFile)
+
+		linkStat := readFileLine(linkStatFile)
+		switch linkStat {
+		case "0":
+			ifaceList[i].LinkStatus = "DOWN"
+		case "1":
+			ifaceList[i].LinkStatus = "UP"
+		default:
+			ifaceList[i].LinkStatus = "UNKNOWN"
```

Comment:
Before, if `err != nil`, this value would not be set to `UNKNOWN`, it would just remain the default value.
Is this what you want to happen now?